src/docfx/config/DependencyPackageUrl.cs

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Newtonsoft.Json;
+
+namespace Microsoft.Docs.Build
+{
+    [JsonConverter(typeof(ShortHandConverter))]
+    internal readonly struct DependencyPackageUrl
+    {
+        public readonly PackageUrl Url;
+
+        public readonly bool InScope;
```

Comment:
```
dependencies:
    _csharplang: 
        remoteUrl: [https://github.com/dotnet/csharplang](https://github.com/dotnet/csharplang)
        branch: master
        inScope: true
```
Can this be serialized?

Reply:
no, unless we change packageurl to class and inherit it, @yufeih  what do you think?

Reply:
add a test for this case?

Reply:
it's not supported, so if we want to support it, we need change `PackageUrl` to class, then I will also change the test correspondingly.

Reply:
or do we really need to support the format above? the user can just specify branch in the fragment of the git url

Reply:
@OsmondJiang , I think turning `PackageUrl` to a class is a good option